cl/corpus_importer/tasks.py

```diff
@@ -1219,21 +1235,228 @@ def make_docket_by_iquery(
+    :param pacer_case_id: The pacer_case_id to use to look up the case
+    :param using: The database to use for the docket lookup
+    :param tag_names: A list of strings that should be added to the docket as
+    tags
+    :param log_results_redis: Log results in redis for the ready mix project
+    :param avoid_trigger_signal: Whether to avoid triggering the iquery sweep
+    signal. Useful for ignoring reports added by the probe daemon or the iquery
+    sweep itself.
+    :return: None if failed, else the ID of the created/updated docket
+    """
+
+    return make_docket_by_iquery_base(
+        self,
+        court_id,
+        pacer_case_id,
+        using,
+        tag_names,
+        log_results_redis,
+        avoid_trigger_signal,
+    )
+
+
+@retry((requests.Timeout, PacerLoginException), tries=3, delay=0.25, backoff=1)
+def query_iquery_page(
+    court_id: str, pacer_case_id: int
+) -> bool | dict[str, Any]:
+    """A small wrapper to query the iquery page for a given PACER case ID to
+    support retries via the @retry decorator in case of a failure.
+
+    :param court_id: A CL court ID where we'll look things up.
+    :param pacer_case_id: The Pacer Case ID to lookup.
+    :return: False if not valid report, otherwise the report.data.
+    """
+
+    report_data = query_case_query_report(court_id, pacer_case_id)
+    if not report_data:
+        logger.info(
+            "No valid data found in iquery page for %s.%s",
+            court_id,
+            pacer_case_id,
+        )
+        return False
+    return report_data
+
+
+@app.task(
+    bind=True,
+    ignore_result=True,
+)
+def probe_iquery_pages(
+    self: Task,
+    court_id: str,
+    testing: bool = False,
+) -> None:
+    """
+    Using the iquery endpoint, to perform forward probing and retrieve the
+    highest watermark we can scrape.
+
+    :param self: The celery task
+    :param court_id: A CL court ID where we'll look things up.
+    :param testing: A boolean indicating whether this was called from tests.
+    :return: None
+    """
+
+    r = get_redis_interface("CACHE")
+    probe_iteration = 1
+    latest_match = 0
+    highest_known_pacer_case_id = int(
+        r.hget("iquery:highest_known_pacer_case_id", court_id) or 0
+    )
+    jitter = compute_binary_probe_jitter(testing)
+    reports_data = []
+    found_match = False
+    while probe_iteration <= settings.IQUERY_PROBE_ITERATIONS:
+        pacer_case_id_to_lookup = compute_next_binary_probe(
+            highest_known_pacer_case_id, probe_iteration, jitter
+        )
+        probe_iteration += 1
+        try:
+            report_data = query_iquery_page(court_id, pacer_case_id_to_lookup)
+        except HTTPError:
+            # Set expiration accordingly and value to 2 to difference from
+            # other waiting times.
+            r.setex(
+                f"iquery:court_wait:{court_id}",
+                settings.IQUERY_COURT_BLOCKED_WAIT,
+                2,
+            )
```

Comment:
is there a reason to use `SETEX` over `SET`? [The documentation](https://redis.io/docs/latest/commands/setex/) for `SETEX` states that it's been deprecated since Redis version 2.6.12.  For setting a key with an expiration time, `SET` is the recommended alternative. 

```suggestion
            r.set(
                f"iquery:court_wait:{court_id}",
                2,
                ex = settings.IQUERY_COURT_BLOCKED_WAIT,
            )
```

Reply:
Yeah, there's no specific reason to use `setex` is the same to use just `set`. I changed it to set, thanks for noticing it.